### PR TITLE
IR-183 Add the concept of an event to the incident report.

### PR DESCRIPTION
**Pull request by Mjwillis**
An incident can be linked to an event and an event can have multiple incidents

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/incidentreporting/jpa/IncidentReport.kt [hunk 1]
```diff
@@ -40,13 +41,17 @@ class IncidentReport(
...
+  var summary: String?,
   var incidentDetails: String,
 
   val reportedBy: String,
   val reportedDate: LocalDateTime,
   @Enumerated(EnumType.STRING)
   var status: IncidentStatus = IncidentStatus.DRAFT,
 
+  @ManyToOne(fetch = FetchType.LAZY, cascade = [CascadeType.ALL])
+  val event: IncidentEvent? = null,
```
Comment (ushkarev) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/incidentreporting/jpa/IncidentReport.kt:53]: what do you think about making this non-nullable? it'd make the data model simpler i think. we could just have the typical case being 1 event ↔︎ 1 incident

Comment (ushkarev) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/incidentreporting/jpa/IncidentReport.kt:53]: the statsy reports / consumers of our data wouldn't have to have branching for orphaned incidents vs those grouped into events

File: src/main/kotlin/uk/gov/justice/digital/hmpps/incidentreporting/jpa/IncidentEvent.kt [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+  @Id
+  @GeneratedValue(strategy = GenerationType.IDENTITY)
+  val id: Long? = null,
+
+  @Column(nullable = false, unique = true, length = 25)
+  val eventId: String,
+  val eventDateAndTime: LocalDateTime,
+  val prisonId: String,
+  var summary: String,
+  var eventDetails: String,
```
Comment (ushkarev) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/incidentreporting/jpa/IncidentEvent.kt:29]: what's the difference between summary and details?

File: src/main/kotlin/uk/gov/justice/digital/hmpps/incidentreporting/jpa/IncidentReport.kt (hunk 1 above)
Comment (Mjwillis) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/incidentreporting/jpa/IncidentReport.kt:53]: Yes could do that - its just I need to create one for all NOMIS migrated records

File: src/main/kotlin/uk/gov/justice/digital/hmpps/incidentreporting/jpa/IncidentEvent.kt (hunk 2 above)
Comment (Mjwillis) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/incidentreporting/jpa/IncidentEvent.kt:29]: Yes not sure there would be one, I've kept it as NOMIS has the concept of title and description and I've reproduced for Event.  Not sure its worth it

**Review by karenmillermoj — APPROVED** [review]
lgtm

**Pull request merged**